File: app/src/main/java/com/iita/akilimo/utils/enums/EnumFieldArea.java
package com.iita.akilimo.utils.enums;

public enum EnumFieldArea {

    UNKNOWN {
        @Override
        public double areaValue() {
            return 0;
        }
    },
    QUARTER_ACRE {
        @Override
        public double areaValue() {
            return 0.25;
        }
    },
    HALF_ACRE {
        @Override
        public double areaValue() {
            return 0.5;
        }
    },
    ONE_ACRE {
        @Override
        public double areaValue() {
            return 1.0;
        }
    },
    ONE_HALF_ACRE {
        @Override
        public double areaValue() {
            return 1.5;
        }
    },
    TWO_HALF_ACRE {
        @Override
        public double areaValue() {
            return 2.5;
        }
    },
    FIVE_ACRE {
        @Override
        public double areaValue() {
            return 5;
        }
    }, EXACT_AREA {
        @Override
        public double areaValue() {
            return -1;
        }
    };

    public abstract double areaValue();
}

File: app/src/main/java/com/akilimo/mobile/utils/enums/EnumFieldArea.kt
package com.akilimo.mobile.utils.enums

enum class EnumFieldArea {
    UNKNOWN {
        override fun areaValue(): Double {
            return 0.0
        }
    },
    QUARTER_ACRE {
        override fun areaValue(): Double {
            return 0.25
        }
    },
    HALF_ACRE {
        override fun areaValue(): Double {
            return 0.5
        }
    },
    ONE_ACRE {
        override fun areaValue(): Double {
            return 1.0
        }
    },
    ONE_HALF_ACRE {
        override fun areaValue(): Double {
            return 1.5
        }
    },
    TWO_HALF_ACRE {
        override fun areaValue(): Double {
            return 2.5
        }
    },
    FIVE_ACRE {
        override fun areaValue(): Double {
            return 5.0
        }
    },
    EXACT_AREA {
        override fun areaValue(): Double {
            return (-1).toDouble()
        }
    };

    abstract fun areaValue(): Double
}
